inventory_reporting/reports/inventory_report.py

```diff
@@ -0,0 +1,334 @@
+from bs4 import BeautifulSoup
+
+from odoo import _, fields, models
+
+
+class InventoryReportXlsx(models.AbstractModel):
+    _name = "report.inventory_reporting.inventory_report_xlsx"
+    _inherit = "report.report_xlsx.abstract"
+
+    def generate_xlsx_report(self, workbook, data, wizard):
+        if data["report_type"] == "valuation":
+            self.generate_valuation_report(workbook, wizard)
+        if data["report_type"] == "storable":
+            self.generate_storable_report(workbook, wizard)
+        elif data["report_type"] == "consumable":
+            self.generate_consumable_report(workbook, wizard)
+
+    def parse_html(self, html_content):
+        if html_content:
+            soup = BeautifulSoup(html_content, "html.parser")
+            return soup.get_text()
+        return False
+
+    def generate_valuation_report(self, workbook, wizard):
+        categories = [
+            "Harness",
+            "PCBA",
+            "PCB",
+            "Electrical",
+            "Component",
+            "Mechanical",
+            "Consumable",
+        ]
```

Comment:
Not a good design to hardcode this (although the client mentioned that hardcoding is acceptable for conditions).  We should probably add a boolean field in the product category to identify the target categories instead.